fix(email-providers): add price to email management page

nightnei opened the pull request:
Related to https://github.com/Automattic/wp-calypso/issues/80967

## Proposed Changes
The issue mentioned above was previously assigned to Letero (partially) and I decided that it's good time to do Letero's part.

With this PR we display renewal price information on the Titan/GW subscription page, to have such information available on this main page of the subscription. 
More details in [the task](https://github.com/Automattic/wp-calypso/issues/80967). 

## Testing Instructions
1) Purchase 2 domains with Titan and GW subscription
2) Open Upgrades->Domains->Select Titan and then GW subscription
3) Assert that you see new descriptions with a price info (Try to turn on/off autorenew)

<table>
<tr>
<td>BEFORE
<td>AFTER
</tr>
<tr>
<td><img width="905" alt="Screenshot 2024-09-11 at 17 18 03" src="https://github.com/user-attachments/assets/26065f96-efdc-48d5-8736-49b53535c2d8">
<td><img width="921" alt="Screenshot 2024-09-11 at 17 07 37" src="https://github.com/user-attachments/assets/b0f6cf02-f4cf-475d-9ca3-ac1e0f7937b1">
</tr>
<tr>
<td><img width="909" alt="Screenshot 2024-09-11 at 17 17 00" src="https://github.com/user-attachments/assets/484e1e91-09f5-4297-ad53-54186f01acaf">
<td><img width="1195" alt="Screenshot 2024-09-11 at 17 10 01" src="https://github.com/user-attachments/assets/8cff4fa5-c982-4414-abd0-19dc15d2b9e9">
</tr>
<tr>
<td><img width="905" alt="Screenshot 2024-09-11 at 17 16 18" src="https://github.com/user-attachments/assets/9ffa90cf-48d6-44f2-8dae-c7b8eab00d52">
<td><img width="909" alt="Screenshot 2024-09-11 at 17 08 45" src="https://github.com/user-attachments/assets/8dcd95d0-7a0d-4d9d-8a5f-97b695feb2a5">
</tr>
<tr>
<td><img width="905" alt="Screenshot 2024-09-11 at 17 15 29" src="https://github.com/user-attachments/assets/eff80da3-c9e4-492f-995a-540becfda2d3">
<td><img width="910" alt="Screenshot 2024-09-11 at 17 13 15" src="https://github.com/user-attachments/assets/1fdb56fb-f781-474e-b76f-7c686d6d2d89">
</tr>
</table>

nightnei committed:
fix(email-providers): add price to email management page

fredrikekelund commented on client/my-sites/email/email-management/home/email-plan-subscription.jsx:
```diff
@@ -35,7 +35,10 @@ class EmailPlanSubscription extends Component {
-				customLabel={ translate( 'Renew now' ) }
+				customLabel={ translate( 'Renew now for {{strong}}%(price)s{{/strong}}', {
+					components: { strong: <strong /> },
+					args: { price: purchase.priceText },
```
This is the price of the previous purchase, right? I don't think we can be sure that the renewal price is the same.

nightnei replied:
I compared with teh "Checkout" page - it's the final renewal price, already with any discounts.

nightnei replied:
<img width="689" alt="Screenshot 2024-09-05 at 14 24 45" src="https://github.com/user-attachments/assets/5833e454-4ed0-4f9b-92c5-b9d3ca80f553">

fredrikekelund commented on client/my-sites/email/email-management/home/email-plan-subscription.jsx:
```diff
@@ -35,7 +35,10 @@ class EmailPlanSubscription extends Component {
 				subscriptionId={ parseInt( purchase.id, 10 ) }
 				tracksProps={ { source: 'email-plan-view' } }
-				customLabel={ translate( 'Renew now' ) }
+				customLabel={ translate( 'Renew now for {{strong}}%(price)s{{/strong}}', {
+					components: { strong: <strong /> },
+					args: { price: purchase.priceText },
+				} ) }
```
```suggestion
```
It looks like `RenewButton` has built-in logic for retrieving the renewal price and rendering almost this exact label. We activate the built-in logic by **not** passing a `customLabel` prop.

`RenewButton` reads the renewal price in almost the same way (which is interesting to know), but I think it's better to rely on that existing logic than reimplementing something similar (if nothing else because `RenewButton` uses `formatCurrency` to format the price.

fredrikekelund reviewed (commented):
Sorry about diving into the technical nitty gritty first and coming back now with more conceptual questions 🤦‍♂️

Anyway, I'm not convinced that this is really better for users. I like what we do on the domain settings page where it says:

> We will attempt to renew your domain on DATE for PRICE

I would prefer doing something similar for email, too. By rendering the price just in the `Renew now` button, it's not 100% clear to users that this is the "canonical" renewal price.

<img width="716" alt="Screenshot 2024-09-09 at 16 45 02" src="https://github.com/user-attachments/assets/06d47c0d-6fa2-4fc5-bcbc-c49a4851734a">

matt-west commented:
I agree with @fredrikekelund that it's not super clear that the price in the button is the regular price and not a special offer. We could make this clearer by adding the price to the label preceding the button.

> Renews on October 22, 2024 for $3.50  [RENEW NOW]

![image](https://github.com/user-attachments/assets/2ed8dd80-b366-42df-904b-54d53a026c0a)

If the user has auto-renewal turned off we should show a different label to reflect that the subscription will expire.

> Expires on October 22, 2024. Renew for $3.50 [RENEW NOW]

![image](https://github.com/user-attachments/assets/f6fed8bf-d422-4e6c-8c3c-53b50d30ffcf)

chad1008 reviewed (commented):
The code changes look good and test well for me, my only concern is the wording of the label when auto renew is turned off.

I agree with @fredrikekelund about the label when auto-renew is turned on. It should describe the anticipated renewal date and price as it currently does 👍 

When auto-renew is off though, having "Renew now for [amount]" in a label next to a "Renew now" button feels repetitive, and the label saying "Renew now" when it's not a clickable action feels a bit off to me. I actually think I'd prefer a combination of the current approach and the original version of this PR:

- if auto renew is **on** render the "Renews on [date] for [amount] label, with a 'renew now' button
- if auto renew is **off** render the "Expires on [date]" label, with a "Renew now for [amount]" button or similar. 

Not a blocker by any means, but to me this would feel clearer and smoother.

fredrikekelund commented:
I agree with @chad1008 regarding the `Expires on […]  Renew for […]  Renew now` scenario being repetitive. We could either move the price to the button, as Chad suggested, or we could just hide the price. That's actually what we do on the domain settings page:

<img width="659" alt="Screenshot 2024-09-13 at 11 39 23" src="https://github.com/user-attachments/assets/a67ab128-67e0-4acb-bf77-b85c98594cf2">

fredrikekelund commented on client/my-sites/email/email-management/home/email-plan-subscription.jsx:
```diff
@@ -76,27 +40,81 @@ class EmailPlanSubscription extends Component {
-			},
-			comment: 'Shows the expiry date of the email subscription',
+
+		const getDescription = () => {
```
I was going to suggest making this a method on the class instead of an inlined function, but I think it'd be even better to:
- convert the file to TS
- make the component into a functional component (for better and easier type support, primarily)

fredrikekelund commented on client/my-sites/email/email-management/style.scss:
```diff
@@ -442,8 +442,16 @@
+	}
+
+	.email-plan-subscription__description-renews-date {
+		margin-left: 20px;
```
```suggestion
		margin-left: 16px;
```
As I mentioned here https://github.com/Automattic/wp-calypso/pull/94239#issuecomment-2348506581, I think we should consider a different approach to displaying the price when auto-renew is off, but I still couldn't help notice that we currently have uneven spacing to the left and right of this text.

fredrikekelund reviewed (commented):
The main thing is figuring out what to display when auto-renew is off. @matt-west, do you have any thoughts based on Chad's and my input? ☝️

matt-west commented:
Thanks for the feedback @chad1008. I agree that it's repetitive.

> We could either move the price to the button, as Chad suggested, or we could just hide the price.

@fredrikekelund Let’s remove the price in this scenario. Hitting the `Renew now` button will take users to the checkout, so they still have a chance to cancel the renewal if needed.

nightnei committed:
refactor: ts, separate component, moved styles to own scss file, removed unused code, default to named exports

nightnei commented on client/my-sites/email/email-management/home/email-plan-subscription/style.scss:
```diff
@@ -0,0 +1,52 @@
+@import "@wordpress/base-styles/breakpoints";
+@import "@wordpress/base-styles/mixins";
+
+.email-plan-subscription__card.card.is-compact {
```
The existing CSS is not perfect, but I decided not to refactor it in this PR. The main thing is - now it's located in a separate file, which is more crucial for the general codebase.

nightnei commented on client/my-sites/email/email-management/home/email-plan-subscription/index.tsx:
```diff
@@ -0,0 +1,110 @@
+					purchase={ purchase }
+					primary={ hasSubscriptionExpired }
+					selectedSite={ selectedSite }
+					subscriptionId={ purchase.id }
```
Removed `pareseInt`, it's useless, we always have int here

nightnei commented on client/my-sites/email/email-management/home/email-plan-header.tsx:
```diff
@@ -63,13 +63,10 @@ const EmailPlanHeader = ( {
 			{ hasEmailSubscription && (
 				<EmailPlanSubscription
 					purchase={ purchase }
-					domain={ domain }
```
We don't use it

nightnei commented on client/my-sites/email/email-management/home/email-plan-header.tsx:
```diff
@@ -19,7 +19,7 @@ type EmailPlanHeaderProps = {
 };
 
-const EmailPlanHeader = ( {
+export const EmailPlanHeader = ( {
```
Added named exports to a few components in this PR. IMO it's awesome rule which we have in Tumblr, there are a lot of pros and no cons.

nightnei commented:
@fredrikekelund Removed price from turned off state

Also:
1) Rewrote to TS
2) Rewrote to functional component
3) Moved styles to own .scss file
4) Created own folder for the component and changed a few default exports to named exports
5) Also a bit removed some unused code

fredrikekelund commented on client/my-sites/email/email-management/home/email-plan-subscription/index.tsx:
```diff
@@ -0,0 +1,110 @@
+
+	const getDescription = () => {
+		const renewalPrice = getRenewalPrice( purchase );
+		const currencyCode = purchase.currencyCode;
+		const formattedRenewalPrice = formatCurrency( renewalPrice, currencyCode, {
+			stripZeros: true,
+		} );
```
```suggestion
		const formattedRenewalPrice = formatCurrency( renewalPrice, purchase.currencyCode, {
			stripZeros: true,
		} );
```
Nit, but I guess we don't need the variable assignment for `currencyCode`

fredrikekelund reviewed (approved):
Nice! LGTM 👍 Thanks for wrapping up this one.

nightnei committed:
chore: remove var

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16944708

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @nightnei for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16944708).
